distributed/client.py

```diff
@@ -2361,7 +2363,10 @@ def get_dataset(self, name, **kwargs):
 
     async def _run_on_scheduler(self, function, *args, wait=True, **kwargs):
         response = await self.scheduler.run_function(
-            function=dumps(function), args=dumps(args), kwargs=dumps(kwargs), wait=wait
+            function=dumps(function, protocol=4),
+            args=dumps(args, protocol=4),
+            kwargs=dumps(kwargs, protocol=4),
```

Comment:
Why does `protocol=4` need to be specified here (and several other places)?

Reply:
Because these pickle.dumps calls aren't regulated by the same handshake context that exists in the normal serialization process.  We don't know anything about the recipient side here, so we remain conservative.  

In practice I think that this is ok.  These arguments are typically small, and this isn't in high-performance codepaths like direct worker-worker communication during execution.